xtask/src/test.rs

```diff
@@ -0,0 +1,36 @@
+use std::fs;
+
+use xshell::{cmd, Shell};
+
+pub fn run() -> anyhow::Result<()> {
+    let sh = Shell::new()?;
+
+    // Compile wasmtime test programs
+    let guard = sh.push_dir("submodules/wasmtime/crates/test-programs");
+    cmd!(sh, "cargo build --target wasm32-wasi --release").run()?;
+    drop(guard);
+
+    for entry in fs::read_dir("submodules/wasmtime/target/wasm32-wasi/release")? {
+        let entry = entry?;
+
+        // skip all files which don't end with `.wasm`
+        if entry.path().extension().map(|p| p.to_str()).flatten() != Some("wasm") {
+            continue;
+        }
+
+        let path = entry.path();
+        let path = match path.to_str() {
+            Some(path) => path,
+            None => continue,
+        };
+
+        // only iterate over the `preview2` test programs
+        if !path.contains("preview2") {
+            continue;
+        }
+
+        cmd!(sh, "src/jco.js run {path}").run()?;
```

Comment:
Are all the test programs commands? Or do we need to invoke the specific test instructions against them?

Reply:
Ah, yeah, so I was going to rewrite this. Right now this just runs every command (many of which are still failing) one-by-one. But what I'd like to do instead is refactor this to _generate_ tests, check them in the repo, and then make them runnable via `cargo test`. That way an individual test failure doesn't mark the end.

But that requires adding a test runner crate, which I think I might just quickly write in a separate PR.